coil-compose-core/src/commonMain/kotlin/coil3/compose/AsyncImagePainter.kt

```diff
@@ -210,47 +211,51 @@ class AsyncImagePainter internal constructor(
         return true
     }
 
-    @OptIn(ExperimentalCoroutinesApi::class)
     override fun onRemembered() = trace("AsyncImagePainter.onRemembered") {
         (painter as? RememberObserver)?.onRemembered()
+        launchJob()
+        isRemembered = true
+    }
 
+    private fun launchJob() {
+        val input = _input ?: return
```

Comment:
Should we add `rememberJob?.cancel()` here? If the `input` changes after we're already remembered and we call `launchJob` I think we want to cancel the previous job.